doc/source/user/numpy-for-matlab-users.rst

```diff
@@ -25,165 +25,51 @@ Some Key Differences
 
 .. list-table::
 
-   * - In MATLAB®, the basic data type is a multidimensional array of
-       double precision floating point numbers.  Most expressions take such
-       arrays and return such arrays.  Operations on the 2-D instances of
-       these arrays are designed to act more or less like matrix operations
-       in linear algebra.
-     - In NumPy the basic type is a multidimensional ``array``.  Operations
-       on these arrays in all dimensionalities including 2D are element-wise
-       operations.  One needs to use specific functions for linear algebra
-       (though for matrix multiplication, one can use the ``@`` operator
-       in python 3.5 and above).
-
-   * - MATLAB® uses 1 (one) based indexing. The initial element of a
+   * - In MATLAB, the basic type is a multidimensional array. By default, MATLAB
+       creates two dimensional arrays of double precision floating point numbers.
+       Operations on the 2D instances of these arrays are designed to act like
+       matrix operations in linear algebra. Most expressions take such arrays and
+       return arrays.
+     - In NumPy, the basic type is a multidimensional ``array``. By default,
+       NumPy creates 0-D or 1-D arrays for inputs of numbers or lists,
```

Comment:
I don't understand "by default." Creating a NumPy array requires calling a routine, which means you have to make some choice.

Reply:
From the sideline: It probably wants to say, unlike matlab where matrices always have two dimensions, NumPy will create 0-D or 1-D arrays unless `ndmin=2` is passed.

Reply:
Good point, I was trying to keep the sentence structure the same the table:

"By default, MATLAB creates two-dimensional arrays of double precision floating point numbers."

and

"By default, NumPy creates 0-D or 1-D arrays for inputs of numbers or lists, respectively."

How about this instead:

"Array assignments in MATLAB are stored as two-dimensional arrays of double precision floating point numbers, unless you specify the number of dimensions and type."

and

"Array assignments in NumPy are stored as N-dimensional arrays with the minimum type required to hold the objects in the sequence, unless you specify the number of dimensions and type."

Reply:
> the minimum type required
Not always true. linspace, for instance, uses floats even when it can use ints. Maybe just "usually" will fix it.

Reply:
> This seems like a deft fix, but
> 
> > the minimum type required
> 
> is not always true. linspace, for instance, uses floats even when it can use ints. Maybe just "usually" will fix it.
> 
> Also
> 
> > two-dimensional
> 
> For consistency at least in this doc, I think we should go with the numeral 2.

OK, change was pushed